node/hack/repair/repair.go

```diff
@@ -0,0 +1,306 @@
+		}
+
+		var before solana.Signature
+
+		decoded, err := hex.DecodeString(emitter)
+		if err != nil {
+			log.Fatalf("Failed to decode emitter address: %v", err)
+		}
+
+		addr := solana.PublicKeyFromBytes(decoded)
+
+		hc := http.Client{Timeout: 10 * time.Second}
+
+		log.Printf("Starting repair for %s (%s)", emitter, addr)
+
+		for {
+			err := limiter.Wait(ctx)
+			if err != nil {
+				log.Fatal(err)
+			}
+
+			limit := 100
+
+			txs, err := sr.GetSignaturesForAddressWithOpts(ctx, addr, &rpc.GetSignaturesForAddressOpts{
+				Limit:  &limit,
+				Before: before,
+			})
+			if err != nil {
+				log.Fatalf("GetConfirmedSignaturesForAddress2 %s: %v", emitter, err)
+			}
+
+			if len(txs) == 0 {
+				log.Printf("fetched all txes for %s (%s)", emitter, addr)
+				break
+			}
+
+			last := txs[len(txs)-1].Signature
+
+			_, lastSeq, err := fetchTxSeq(ctx, sr, last)
+			if err != nil {
+				log.Fatalf("fetch last tx seq: %v", err)
+			}
+
+			_, firstSeq, err := fetchTxSeq(ctx, sr, txs[0].Signature)
+			if err != nil {
+				log.Fatalf("fetch first tx seq: %v", err)
+			}
+
+			log.Printf("fetched %d transactions, from %s (%d) to %s (%d)",
+				len(txs), txs[0].Signature, firstSeq, last, lastSeq)
+
+			if highest < lastSeq {
+				log.Printf("skipping (%d < %d)", highest, lastSeq)
+				goto skip
+			}
+			if lowest > firstSeq {
+				log.Printf("done (%d < %d)", lowest, lastSeq)
+				break
+			}
+			for _, p := range msgs {
+				if p.Sequence > lastSeq && p.Sequence < firstSeq {
+					offset := firstSeq - p.Sequence
+					log.Printf("repairing: %d (offset %d)", p.Sequence, offset)
+
+					var tx *rpc.TransactionWithMeta
+					var nseq uint64
+					var err error
+
+					for {
+						tx, nseq, err = fetchTxSeq(ctx, sr, txs[offset].Signature)
+						if err != nil {
+							log.Fatalf("failed to fetch %s at offset %d: %v", txs[offset].Signature, offset, err)
+						}
+						if tx == nil {
+							offset += 1
+							log.Printf("not a Wormhole tx, offset +1")
+							time.Sleep(1 * time.Second)
+							continue
+						}
+						if nseq != p.Sequence {
+							log.Printf("%d != %d, offset +%d", nseq, p.Sequence, nseq-p.Sequence)
+							offset += nseq - p.Sequence
+							time.Sleep(1 * time.Second)
+							continue
+						} else {
+							break
+						}
+					}
+
+					acc, err := process(tx)
+					if err != nil {
+						log.Fatalf("process: %v", err)
+					}
+
+					log.Printf("found account %v (%s)", acc, hex.EncodeToString(acc[:]))
+
+					_, err = admin.SendObservationRequest(ctx, &nodev1.SendObservationRequestRequest{
+						ObservationRequest: &gossipv1.ObservationRequest{
+							ChainId: uint32(vaa.ChainIDSolana),
+							TxHash:  acc[:],
+						}})
+					if err != nil {
+						log.Fatalf("SendObservationRequest: %v", err)
+					}
+
+					for {
+						log.Printf("verifying %d", p.Sequence)
+						req, err := http.NewRequestWithContext(ctx, "GET", fmt.Sprintf(
+							"%s/v1/signed_vaa/%d/%s/%d",
+							publicRPCEndpoints[0],
```

Comment:
It's a shame we don't retry this to a node other than the certus node here, but I get why you might not want to do that.

Reply:
Actually... we could, but I wanted to avoid picking a random node on the first try that wasn't available

Reply:
Works for me.